### PR TITLE
feat: limited number of confirmation skips

### PR DESCRIPTION
**Pull request by 0xbrayo**
closes https://github.com/ErikBjare/gptme/issues/332

> [!IMPORTANT]
> Adds limited confirmation skips to `ask_execute()` using `auto_skip_count` in `ask_execute.py`.
> 
>   - **Behavior**:
>     - Introduces `auto_skip_count` in `ask_execute.py` to limit the number of confirmation skips.
>     - Modifies `ask_execute()` to decrement `auto_skip_count` when skipping confirmation.
>     - Uses regex to parse "auto" command with optional skip count in `ask_execute()`.
>   - **Misc**:
>     - Imports `re` module in `ask_execute.py` for regex operations.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ErikBjare%2Fgptme&utm_source=github&utm_medium=referral)<sup> for c035a16264fd8b20fa33132dc22028372362a680. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by 0xbrayo** [comment]
closes #332

File: gptme/util/ask_execute.py [hunk 1]
```diff
@@ -128,9 +132,14 @@ def ask_execute(question="Execute code?", default=True) -> bool:
                 return ask_execute("Execute with changes?", default)
             return False
 
-    # secret option to stop asking for the rest of the session
-    if answer == "auto":
-        return (override_auto := True)
+    # secret option to skip asking for a number of times
+    match = re.match(r'(auto)(?: (\d+))?', answer)
```
[the remark this answers is not in the document]
Comment (0xbrayo) [reply gptme/util/ask_execute.py]: Don't think it's necessary really. user might type a space before auto

[the remark this answers is not in the document]
Comment (0xbrayo) [reply gptme/util/ask_execute.py]: Also compiled the regex

File: gptme/util/ask_execute.py [hunk 2]
```diff
@@ -22,13 +23,17 @@
...
+auto_skip_count = 0
 copiable = False
 editable = False
 
 # Editable text state
 _editable_text = None
 _editable_ext = None
 
+#Compiled regex pattern
+pattern = re.compile(r'(auto)(?: (\d+))?')
```
Comment (ErikBjare) [inline gptme/util/ask_execute.py]: Call this something like `re_auto`, I don't think we need to compile it, at least not at import-time.

File: gptme/util/ask_execute.py [hunk 3]
```diff
@@ -128,9 +135,14 @@ def ask_execute(question="Execute code?", default=True) -> bool:
                 return ask_execute("Execute with changes?", default)
             return False
 
-    # secret option to stop asking for the rest of the session
-    if answer == "auto":
-        return (override_auto := True)
+    # secret option to skip asking for a number of times
```
Comment (ErikBjare) [inline gptme/util/ask_execute.py]: Not so secret anymore
```suggestion
    # skip asking for a number of times
```

**Comment by ErikBjare** [comment]
Nice!

**Comment by 0xbrayo** [comment]
Forgot to [auto] as an option displayed in confirmation prompt. Guess it
remains secret?

**Comment by 0xbrayo** [comment]
*add

**Pull request merged**